### PR TITLE
removed double loop in app.py / added method for plotting mincircles.…

### PR DESCRIPTION
**Pull request by JmzCS50**
removed double loop in app.py / added method for plotting mincircles.py / added geojson of US map to plot on(good for future testing)

### CODE REVIEW

File: MinimalCirclesPath.py [hunk 1]
```diff
@@ -1,4 +1,9 @@
 from math import sin, cos, radians, asin, atan2, sqrt, degrees, ceil, floor
+import matplotlib.pyplot as plt
+import geopandas as gpd
```
Comment (Gabrieldowen) [inline MinimalCirclesPath.py:3]: should we add these to the requirements.txt? I know you dont need it for the actual app but we may still want it there

**Review by Gabrieldowen — COMMENTED** [review]
other than possibly adding the library to the requirements i love it. I think this would be a good thing to show in our final presentation as like a "here the points it calls from with x radius and from there we filter them down and..."

**Review by emilyridge — APPROVED** [review]
Looks good to me!

File: MinimalCirclesPath.py [hunk 2]
```diff
@@ -94,13 +99,49 @@ def getPath( startLat, startLong, destLat, destLong, radius, pathWidth):
     return coordList
        
 
-
+def plot_path_on_us_map(coordList):
+    # Load the US map from the GeoJSON file
+    us_map = gpd.read_file('us.geojson')
```
Comment (n-grounds) [inline MinimalCirclesPath.py:104]: isn't this already read once and stored in the `gdf` variable?

File: MinimalCirclesPath.py [hunk 3]
```diff
@@ -94,13 +99,49 @@ def getPath( startLat, startLong, destLat, destLong, radius, pathWidth):
...
+    # Show the plot
+    plt.show()
 
 # THIS IS AN EXAMPLE
-# uncomment and run `python3 MinimalCirclesPath.py` to see example of what getPath() returns
+# uncomment and run `python3 MinimalCirclesPath.py` to see example of what getPath() and plot_path_on_us_map(pathList) returns
 """
 if __name__ == '__main__':
-    pathList = getPath(startLat = 32.7767, startLong = 96.7970, destLat = 39.7392, destLong = 104.9903, radius = 100,  pathWidth = 50)
+    pathList = getPath(startLat = 32.8968,  # DFW Latitude
```
Comment (n-grounds) [inline MinimalCirclesPath.py:137]: nice to have hard-coded values as examples, but what about checking `sys.argv` and if the user supplied 4 arguments treat them as lat/long coordinates to plot, e.g.,
```python
import sys
startLat, startLong, destLat, destLong = 32.8968, -97.0380, 33.9416, -118.4085
if len( sys.argv ) == 4:
    startLat, startLong, destLat, destLong = [ int( a ) for a  in sys.argv ]
```

File: MinimalCirclesPath.py (hunk 1 above)
Comment (JmzCS50) [reply MinimalCirclesPath.py:3]: sure we can do that

File: MinimalCirclesPath.py (hunk 2 above)
Comment (JmzCS50) [reply MinimalCirclesPath.py:104]: > isn't this already read once and stored in the `gdf` variable?

correct, will fix

File: MinimalCirclesPath.py [hunk 4]
```diff
@@ -94,13 +99,49 @@ def getPath( startLat, startLong, destLat, destLong, radius, pathWidth):
...
+    plt.title('Path Visualization on US Map')
+    plt.xlabel('Longitude')
+    plt.ylabel('Latitude')
+    
+    # Show the plot
+    plt.show()
 
 # THIS IS AN EXAMPLE
-# uncomment and run `python3 MinimalCirclesPath.py` to see example of what getPath() returns
+# uncomment and run `python3 MinimalCirclesPath.py` to see example of what getPath() and plot_path_on_us_map(pathList) returns
```
[the remark this answers is not in the document]
Comment (JmzCS50) [reply MinimalCirclesPath.py:134]: these are there for testing the functions locally and not for someone to run it by using MinimalCirclesPath.

**Pull request merged**